### PR TITLE
Reduction of lambda role S3 permissions

### PR DESCRIPTION
**Pull request by gaborschulz-aws**
*Issue #, if available:*

*Description of changes:*
- Fixed documentation typo
- Removed redundant code
- Backported reduction of lambda role S3 permissions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: static/Cost/300_Optimization_Data_Collection/Code/module-ecs-chargeback.yaml [hunk 1]
```diff
@@ -88,7 +91,10 @@ Resources:
                   - "sqs:ReceiveMessage"
                   - "sqs:DeleteMessage"
                   - "sqs:GetQueueAttributes"
-                Resource: !Sub "arn:aws:sqs:${AWS::Region}:${AWS::AccountId}:EcsChargebackModule-TaskQueue"
+                Resource: 
+                  - !Sub "arn:aws:sqs:${AWS::Region}:${AWS::AccountId}:EcsChargebackModule-TaskQueue"
+                  - !Ref DestinationBucketARN
+                  - !Sub "${DestinationBucketARN}/*"
```
Comment (iakov-aws) [inline static/Cost/300_Optimization_Data_Collection/Code/module-ecs-chargeback.yaml:97]: Wouldm make sense to split to 2 policiies?

File: static/Cost/300_Optimization_Data_Collection/Code/module-rds-usage.yaml [hunk 2]
```diff
@@ -394,7 +396,7 @@ Resources:
       DatabaseName: !Ref DatabaseName
       Targets:
         S3Targets:
-          - Path: !Sub "s3://${DestinationBucket}/${CFDataName}/${CFDataName}-data/"
+          - Path: !Sub "s3://${DestinationBucket}/${CFDataName}/rds-usage-data/"
```
Comment (iakov-aws) [inline static/Cost/300_Optimization_Data_Collection/Code/module-rds-usage.yaml:399]: maybe rather change CFDataName default?

File: static/Cost/300_Optimization_Data_Collection/Code/module-ecs-chargeback.yaml (hunk 1 above)
Comment (gaborschulz-aws) [reply static/Cost/300_Optimization_Data_Collection/Code/module-ecs-chargeback.yaml:97]: Done

File: static/Cost/300_Optimization_Data_Collection/Code/module-rds-usage.yaml (hunk 2 above)
Comment (gaborschulz-aws) [reply static/Cost/300_Optimization_Data_Collection/Code/module-rds-usage.yaml:399]: Alternatively, I can modify the lambda function to use `CFDataName` as well

Comment (iakov-aws) [reply static/Cost/300_Optimization_Data_Collection/Code/module-rds-usage.yaml:399]: this must be aligned, but probably not now

**Review by iakov-aws — APPROVED** [review]

File: static/Cost/300_Optimization_Data_Collection/Code/lambda_s3_athen_cf_template.yaml [hunk 3]
```diff
@@ -52,7 +52,7 @@ Resources:
                 - lambda.amazonaws.com
         Version: 2012-10-17
       ManagedPolicyArns:
-        - arn:aws:iam::aws:policy/AWSLambdaExecute
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
```
Comment (lpmi-13) [inline static/Cost/300_Optimization_Data_Collection/Code/lambda_s3_athen_cf_template.yaml:55]: 👏

**Pull request merged**